scripts/host-check

```diff
@@ -1126,31 +1140,32 @@ def check_iommu() -> CheckFuncReturn:
         return (
             CheckState.ERROR,
             f"The cmd {cmd!r} failed - unable to\n"
-            "determine the network devices on the host. IOMMU is enabled.",
-        )
-    net_iommu_devices = network_devices & set(iommu_devices)
-    if not net_iommu_devices:
-        return (
-            warning_state,
-            "IOMMU enabled for vfio-pci, but no network PCI devices found.",
-        )
-    net_iommu_devs_list = []
-    for device in sorted(net_iommu_devices):
-        net_iommu_devs_list.append(
-            net_devices_dict[device] + " (" + device + ")"
+            "determine the network devices on the host.",
         )
+
+    # If in IOMMU mode then check if the network devices are in the IOMMU group
+    if iommu_devices:
+        network_devices &= set(iommu_devices)
+        if not network_devices:
+            return (
+                CheckState.WARNING,
+                "No PCI network devices found with IOMMU enabled for vfio-pci.",
```

Comment:
How do we know we are using vfio-pci?

Reply:
With other changes to check_pci_devices, this is now indicated by require_iommu flag